sql/server.go

```diff
@@ -139,7 +198,10 @@ func (s *Server) exec(req driver.Request) (driver.Response, error) {
 		return resp, err
 	}
 	for _, stmt := range stmts {
-		// TODO(pmattis): Fill placeholders.
+		// Bind all the placeholder variables in the stmt to actual values.
+		if err = s.bindValues(stmt, parameters{params: req.Params}); err != nil {
```

Comment:
We bind the values here. Another option is for us to bind them during expression evaluation. I suppose one advantage of doing it here is that it's done before the query execution plan is mapped out. Another possibility is to do it earlier while parsing.

A big problem of the current approach is that it needs to find all the expressions in the AST and it varies while we add new expressions to the AST.